src/common/templates/_monitoring.tpl

```diff
@@ -43,3 +43,32 @@ prometheus.io/scrape: {{ $enabled | quote}}
   protocol: "TCP"
 {{- end }}
 {{- end -}}
+
+{{/* Podmonitor template to be added for Google Managed prometheus
+{{ include "harnesscommon.monitoring.podMonitor" (dict "name" "serviceName" "ctx" $) }}
+*/}}
+{{- define "harnesscommon.monitoring.podMonitor" -}}
+{{- $ := .ctx }}
+{{- $enabled := and .ctx.Values.global.monitoring.enabled (eq .ctx.Values.global.monitoring.managedPlatform "google") -}}
+{{- $localMonitoring := default (dict) ((pluck "monitoring" .ctx.Values) | first) -}}
+{{- $globalMonitoring := default (dict) ((pluck "monitoring" .ctx.Values.global) | first) -}}
+{{- $monitoring := (mergeOverwrite $globalMonitoring $localMonitoring ) }}
+{{- $port := (pluck "port" $monitoring) | first }}
+{{- $path := (pluck "path" $monitoring) | first }}
+{{- $namespace := $.Release.Namespace }}
+{{- if $enabled -}}
+apiVersion: monitoring.googleapis.com/v1
+kind: PodMonitoring
+metadata:
+  name: {{ $.Chart.Name }}
+  namespace:  {{ $namespace }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ default $.Chart.Name .name }}
```

Comment:
this should be same as the one being used by the service's Deployment. Also the whole code should either use `.ctx` or `$` as both mean the same thing above.